models/experimental.py

```diff
@@ -74,8 +74,9 @@ def attempt_load(weights, device=None, inplace=True, fuse=True):
     from models.yolo import Detect, Model
 
     model = Ensemble()
-    if ".zip" in weights:
+    if isinstance(weights, str) and ".zip" in weights:
```

Comment:
```suggestion
    if isinstance(weights, str) and weights.endswith(".zip"):
```
是不是好一点，这和下一行的 `weights = weights[:-4]` 逻辑就匹配一致。

如果是 `".zip" in weights` 做检查的化，下一句是不是应该是 `weights = weitghts.replace(".zip", "")

Reply:
赞同，可以改一下。 @ccssu